js/views/conversation_view.js

```diff
@@ -620,7 +627,10 @@
         messageDetail: function(e, data) {
             var view = new Whisper.MessageDetailView({
                 model: data.message,
-                conversation: this.model
+                conversation: this.model,
+                // we pass these in to allow nested panels
+                listenBack: this.listenBack.bind(this),
+                resetPanel: this.resetPanel.bind(this)
```

Comment:
Do you think that moving to a popup dialog like Android would make things easier to understand?

Alternatively, one way to improve things here would be to replicate the message that failed to send somewhere on the three-button warning page to make it clear what 'Send anyway' applies to.